testsuite/tests/typing-func-subtype/opt_arg_subtyping.ml

```diff
@@ -0,0 +1,48 @@
+(* TEST
+ expect;
+*)
+
+[@@@warning "-16"]
+
+let f ?x:_ _ = assert false;;
+
+(* This should be fine. There's a positional argument next, so we know that
+erasing the optional argument is okay. *)
+let _ = (f : 'a -> 'a);;
+
+[%%expect{|
+val f : ?x:'a -> 'b -> 'c = <fun>
+- : 'a -> 'a = <fun>
+|}];;
+
+let f ?x:_ ~y:_ = 1;;
+
+(* This shouldn't be allowed - we're erasing an unerasable optional argument. *)
+let _ = (f : y:'a -> int);;
+
+[%%expect {|
+val f : ?x:'a -> y:'b -> int = <fun>
+Line 4, characters 9-10:
+4 | let _ = (f : y:'a -> int);;
+             ^
+Error: The value "f" has type "?x:'b -> y:'c -> int"
+       but an expression was expected of type "y:'a -> int"
+       Labels "?x" and "y" do not match
```

Comment:
Would be nice if it's possible to say something like this (maybe wording needs to be modified to match the correct constraint)
```suggestion
Error: The value "f" has type "?x:'b -> y:'c -> int"
       but an expression was expected of type "y:'a -> int"
       The optional argument "?x" cannot be erased:
       It is not followed by an unlabeled argument.
```

Or maybe `The last erased argument is not followed by an unlabeled argument.`

Reply:
That makes sense, but after some investigation I think it's a more involved change - perhaps a separate PR? The check is currently done in Typecore but the errors are raised in Ctype. If someone wants to give a bit of guidance on the right way to bridge that gap I could take a crack at it but I don't think I currently have enough context on this codebase to take a reasonable approach.